### DISCUSSION TITLE
Set start at item number or not fail on errors.

### DISCUSSION THREAD

**Discussion by timcanning** [Barbarur/NovaPoint#10] [Ideas]
When the process (remove version history in this case) fails, due to a corrupt file for instance, it would be good to have a way to set the item count to start again? Alternatively, not fail on errors if possible and just log them.

**Comment by Barbarur** [comment]
Hi @timcanning ,
Thank you for posting.

Can you explain more on where if fails and doesn't continue?

When deleting all version, if there is an error during the deletion, the error is recorded and the process jumps to the next file.
When keeping specific number of versions, versions are deleted one by one. If there is an error deleting a specific version, that is recorded and it continues to delete the next version.

Are you expriencing a different behavior?

**Comment by timcanning** [comment]
Howdy,


Thanks for getting back to me.

So this document library has about 300k files. It has been there for several years and I think it is hitting corrupt files and then bailing out.

This is the third run today, and it appears that I take a copy of the file, delete the original and start again and off it goes.

Let me know if you need any further information.

[cid:741b9248-f18d-4a9b-8ebc-5d82501255d2]



Tim Canning     Infrastructure Architect

[cid:Red21_logo_SIG2-Copy_7a1df1cb-0ccb-40d8-9f38-4933c30ab3bc.png]
RED21   Getting IT Right
WWW.RED21.COM.AU<http://www.RED21.com.au>

[cid:twitter60v4_738510bc-8a55-4c8e-8831-4c0f1d4c98e7.png]<http://twitter.com/red21au>  [cid:linkedin60v4_a7efd89b-d953-42de-b20f-1023b7178ee9.png] <http://www.linkedin.com/company/red21>   [cid:facebook60v4_76cebc4d-259e-4085-81ec-5281482d52ee.png] <http://www.facebook.com/RED21.com.au>






Confidentiality and Disclaimer:
The contents of this e-mail are confidential and are intended for the named recipient(s) only. If you are not the intended recipient you are hereby notified that any use, disclosure or distribution of the information contained in this e-mail is prohibited. If you have received this in error, please notify the sender immediately, and delete the e-mail and any attachments. RED21 has implemented anti-virus software and whilst all care is taken, it is the recipient’s responsibility to ensure that the message and any attachments are scanned for viruses prior to use.



________________________________

**Comment by Barbarur** [comment]
Hello @timcanning ,

If you click on the solution folder, you should see the .txt file with the logs. They will be too long to share them all, but please share the top part where we can see the parameters added when running the Solution, and the end where we see the error exception message.

I’ve reviewed the code, I haven’t seen how an error could break the run without continuing with the next file.

**Comment by timcanning** [comment]
Howdy,

Thanks for the follow up.


Last run (I am running this about twice a day until we get through all 300k files).

Over the last week the point at which it stops with this type of error has been slowly increasing, and is getting up to 150k items.



Attached is the full logged.

Here is the last page of the last run log and the corresponding screenshot of where it had the error and stopped between 145k and 150k items.


2024/05/19 16:21:49 - [AppInfo] - Got access token from memory
2024/05/19 16:21:49 - [AppInfo] - Access Token expiration time: 19/05/2024 5:07:59 PM +00:00
2024/05/19 16:21:50 - [RemoveFileVersionAuto] - Deleting all version 'vecteezy_new-year-illustration_dp1220.eps'
2024/05/19 16:21:50 - [RemoveFileVersionAuto] - Start processing File 'vecteezy_new-year-illustration_dp1220.jpg'
2024/05/19 16:21:50 - [AppInfo] - Getting Access Token for root site https://facciaus.sharepoint.com
2024/05/19 16:21:50 - [AppInfo] - Got access token from memory
2024/05/19 16:21:50 - [AppInfo] - Access Token expiration time: 19/05/2024 5:07:59 PM +00:00
2024/05/19 16:21:50 - [RemoveFileVersionAuto] - Deleting all version 'vecteezy_new-year-illustration_dp1220.jpg'
2024/05/19 16:21:50 - [RemoveFileVersionAuto] - Start processing File '19406363.eps'
2024/05/19 16:21:50 - [AppInfo] - Getting Access Token for root site https://facciaus.sharepoint.com
2024/05/19 16:21:50 - [AppInfo] - Got access token from memory
2024/05/19 16:21:50 - [AppInfo] - Access Token expiration time: 19/05/2024 5:07:59 PM +00:00
2024/05/19 16:21:50 - [RemoveFileVersionAuto] - Deleting all version '19406363.eps'
2024/05/19 16:21:50 - [RemoveFileVersionAuto] - Start processing File '19406363.jpg'
2024/05/19 16:21:50 - [AppInfo] - Getting Access Token for root site https://facciaus.sharepoint.com
2024/05/19 16:21:50 - [AppInfo] - Got access token from memory
2024/05/19 16:21:50 - [AppInfo] - Access Token expiration time: 19/05/2024 5:07:59 PM +00:00
2024/05/19 16:21:50 - [RemoveFileVersionAuto] - Deleting all version '19406363.jpg'
2024/05/19 16:21:50 - [RemoveFileVersionAuto] - Start processing File '4723946.eps'
2024/05/19 16:21:50 - [AppInfo] - Getting Access Token for root site https://facciaus.sharepoint.com
2024/05/19 16:21:50 - [AppInfo] - Got access token from memory
2024/05/19 16:21:50 - [AppInfo] - Access Token expiration time: 19/05/2024 5:07:59 PM +00:00
2024/05/19 16:21:50 - [RemoveFileVersionAuto] - Deleting all version '4723946.eps'
2024/05/19 16:21:50 - [RemoveFileVersionAuto] - Start processing File '4721406.jpg'
2024/05/19 16:21:50 - [AppInfo] - Getting Access Token for root site https://facciaus.sharepoint.com
2024/05/19 16:21:50 - [AppInfo] - Got access token from memory
2024/05/19 16:21:50 - [AppInfo] - Access Token expiration time: 19/05/2024 5:07:59 PM +00:00
2024/05/19 16:21:50 - [RemoveFileVersionAuto] - Deleting all version '4721406.jpg'
2024/05/19 16:21:50 - [RemoveFileVersionAuto] - Start processing File '4723947.ai'
2024/05/19 16:21:50 - [AppInfo] - Getting Access Token for root site https://facciaus.sharepoint.com
2024/05/19 16:21:50 - [AppInfo] - Got access token from memory
2024/05/19 16:21:50 - [AppInfo] - Access Token expiration time: 19/05/2024 5:07:59 PM +00:00
2024/05/19 16:21:50 - [RemoveFileVersionAuto] - Deleting all version '4723947.ai'
2024/05/19 16:21:50 - [RemoveFileVersionAuto] - Start processing File 'GMT20201201-003143_My-Allianc_gallery_3840x2160.mp4'
2024/05/19 16:21:50 - [AppInfo] - Getting Access Token for root site https://facciaus.sharepoint.com
2024/05/19 16:21:50 - [AppInfo] - Got access token from memory
2024/05/19 16:21:50 - [AppInfo] - Access Token expiration time: 19/05/2024 5:07:59 PM +00:00
2024/05/19 16:21:50 - [RemoveFileVersionAuto] - Deleting all version 'GMT20201201-003143_My-Allianc_gallery_3840x2160.mp4'
2024/05/19 16:21:50 - [NPLogger] - Error processing Item '/Sydney Share/TRADE, SERVICES & HOSTING/SERVICES/Zoom Hosting/201130 Oncord Training_AllianceFrancaise/Day 2/Day2_PlenAdvanced/GMT20201201-003143_My-Allianc_gallery_3840x2160.mp4'
2024/05/19 16:21:50 - [NPLogger] - Exception: Unable to cast object of type 'System.String' to type 'Microsoft.SharePoint.Client.FieldLookupValue'.
2024/05/19 16:21:50 - [NPLogger] - Trace:    at NovaPointLibrary.Solutions.Automation.RemoveFileVersionAuto.AddRecord(String siteUrl, List oList, ListItem oItem, String versionsDeletedCount, String versionsDeletedMB, String remarks)
   at NovaPointLibrary.Solutions.Automation.RemoveFileVersionAuto.RemoveFileVersions(String siteUrl, List oList, ListItem oItem)
   at NovaPointLibrary.Solutions.Automation.RemoveFileVersionAuto.ProcessItems(String siteUrl, List oList, ProgressTracker parentProgress)
2024/05/19 16:21:50 - [NPLogger] - Error processing DocumentLibrary '/Sydney Share/Forms/AllItems.aspx'
2024/05/19 16:21:50 - [NPLogger] - Exception: Unable to cast object of type 'System.String' to type 'Microsoft.SharePoint.Client.FieldLookupValue'.
2024/05/19 16:21:50 - [NPLogger] - Trace:    at NovaPointLibrary.Solutions.Automation.RemoveFileVersionAuto.AddRecord(String siteUrl, List oList, ListItem oItem, String versionsDeletedCount, String versionsDeletedMB, String remarks)
   at NovaPointLibrary.Solutions.Automation.RemoveFileVersionAuto.ProcessItems(String siteUrl, List oList, ProgressTracker parentProgress)
   at NovaPointLibrary.Solutions.Automation.RemoveFileVersionAuto.ProcessItems(String siteUrl, List oList, ProgressTracker parentProgress)
   at NovaPointLibrary.Solutions.Automation.RemoveFileVersionAuto.RunScriptAsync()
2024/05/19 16:21:50 - [SPOSiteCollectionAdminCSOM] - Processing 'https://facciaus.sharepoint.com' setting ***@***.***' IsSiteAdmin 'False'
2024/05/19 16:21:50 - [AppInfo] - Getting Access Token for root site https://facciaus-admin.sharepoint.com
2024/05/19 16:21:50 - [AppInfo] - Finish aquiring new Access Token with Refresh Token.
2024/05/19 16:21:50 - [AppInfo] - Access Token expiration time: 19/05/2024 5:48:41 PM +00:00
2024/05/19 16:21:51 - [NPLogger] - COMPLETED: Solution has finished correctly!


[cid:49c46890-6463-4178-88c9-e850a06867ed]



Tim Canning     Infrastructure Architect

[cid:Red21_logo_SIG2-Copy_7a1df1cb-0ccb-40d8-9f38-4933c30ab3bc.png]
RED21   Getting IT Right
WWW.RED21.COM.AU<http://www.RED21.com.au>

[cid:twitter60v4_738510bc-8a55-4c8e-8831-4c0f1d4c98e7.png]<http://twitter.com/red21au>  [cid:linkedin60v4_a7efd89b-d953-42de-b20f-1023b7178ee9.png] <http://www.linkedin.com/company/red21>   [cid:facebook60v4_76cebc4d-259e-4085-81ec-5281482d52ee.png] <http://www.facebook.com/RED21.com.au>






Confidentiality and Disclaimer:
The contents of this e-mail are confidential and are intended for the named recipient(s) only. If you are not the intended recipient you are hereby notified that any use, disclosure or distribution of the information contained in this e-mail is prohibited. If you have received this in error, please notify the sender immediately, and delete the e-mail and any attachments. RED21 has implemented anti-virus software and whilst all care is taken, it is the recipient’s responsibility to ensure that the message and any attachments are scanned for viruses prior to use.



________________________________

**Comment by Barbarur** [comment]
Hello @timcanning ,

Thank you for the information. I think I found the issue. It seems the solution crash when adding the record on the report, not while deleting versions. The property of the total storage for the affected files is read incorrectly.

If you go to the Site Storage Metrics and check for the file that crashed *'/Sydney Share/TRADE, SERVICES & HOSTING/SERVICES/Zoom Hosting/201130 Oncord Training_AllianceFrancaise/Day 2/Day2_PlenAdvanced/GMT20201201-003143_My-Allianc_gallery_3840x2160.mp4'*.

Are you able to see the total storage of that file?

**Comment by TheRealStriff** [comment]
I have similar issue, when it fails, it stops and does not continue.
But the issue for me, is not corrupted files, it seems cannot handle files with large amount of versions (over 2000 versions on some files)

I've logged this under "issues"

**Reply by Barbarur** [reply]
Hi @TheRealStriff,

I've checked the logs you left on the issues list. They looks the same as Tim's. The solution breaks when casting a String into FieldLookupValue when adding the record on the .CSV report.

If you have any other logs that shows different exception message, please share them.

**Reply by TheRealStriff** [reply]
Looks like same issue/
Tried different site. Fails on MP4 file which has no versions.
file size reported in the library - 2.05GB
Storage metrics - 2.1 GB



`2024/05/22 22:56:39 - [AppInfo] - Got access token from memory
2024/05/22 22:56:39 - [AppInfo] - Access Token expiration time: 23/05/2024 12:09:08 AM +00:00
2024/05/22 22:56:39 - [NPLogger] - Error processing Item '/sites/LakeV/SRSafety/05. Event Management/2023/03 March/20230301 EX69 Fell from Bench/20230301 EX69 Fell off Bench/02. Event Photos, Videos & Recordings/UAV Video/DJI_20230301072633_0099_V.MP4'
2024/05/22 22:56:39 - [NPLogger] - Exception: Unable to cast object of type 'System.String' to type 'Microsoft.SharePoint.Client.FieldLookupValue'.
2024/05/22 22:56:39 - [NPLogger] - Trace:    at NovaPointLibrary.Solutions.Automation.RemoveFileVersionAuto.AddRecord(String siteUrl, List oList, ListItem oItem, String versionsDeletedCount, String versionsDeletedMB, String remarks)
   at NovaPointLibrary.Solutions.Automation.RemoveFileVersionAuto.RemoveFileVersions(String siteUrl, List oList, ListItem oItem)
   at NovaPointLibrary.Solutions.Automation.RemoveFileVersionAuto.ProcessItems(String siteUrl, List oList, ProgressTracker parentProgress)
2024/05/22 22:56:39 - [NPLogger] - Error processing DocumentLibrary '/sites/LakeV/SRSafety/Forms/AllItems.aspx'
2024/05/22 22:56:39 - [NPLogger] - Exception: Unable to cast object of type 'System.String' to type 'Microsoft.SharePoint.Client.FieldLookupValue'.
2024/05/22 22:56:39 - [NPLogger] - Trace:    at NovaPointLibrary.Solutions.Automation.RemoveFileVersionAuto.AddRecord(String siteUrl, List oList, ListItem oItem, String versionsDeletedCount, String versionsDeletedMB, String remarks)
   at NovaPointLibrary.Solutions.Automation.RemoveFileVersionAuto.ProcessItems(String siteUrl, List oList, ProgressTracker parentProgress)
   at NovaPointLibrary.Solutions.Automation.RemoveFileVersionAuto.ProcessItems(String siteUrl, List oList, ProgressTracker parentProgress)
   at NovaPointLibrary.Solutions.Automation.RemoveFileVersionAuto.RunScriptAsync()
2024/05/22 22:56:39 - [SPOSiteCollectionAdminCSOM] - Processing 'https://o365xxxx.sharepoint.com/sites/LakeV setting 'xxxadmin@o365XXXXonmicrosoft.com' IsSiteAdmin 'False'
2024/05/22 22:56:39 - [AppInfo] - Getting Access Token for root site https://o365xxxx-admin.sharepoint.com
2024/05/22 22:56:39 - [AppInfo] - Got access token from memory
2024/05/22 22:56:39 - [AppInfo] - Access Token expiration time: 23/05/2024 12:20:34 AM +00:00
2024/05/22 22:56:39 - [NPLogger] - COMPLETED: Solution has finished correctly!`

**Comment by timcanning** [comment]
Not 100% sure this is the correct information, but take a look and let me know.

[cid:cef9b54a-534a-4161-a508-9228490d8654]



Tim Canning     Infrastructure Architect

[cid:Red21_logo_SIG2-Copy_7a1df1cb-0ccb-40d8-9f38-4933c30ab3bc.png]
RED21   Getting IT Right
WWW.RED21.COM.AU<http://www.RED21.com.au>

[cid:twitter60v4_738510bc-8a55-4c8e-8831-4c0f1d4c98e7.png]<http://twitter.com/red21au>  [cid:linkedin60v4_a7efd89b-d953-42de-b20f-1023b7178ee9.png] <http://www.linkedin.com/company/red21>   [cid:facebook60v4_76cebc4d-259e-4085-81ec-5281482d52ee.png] <http://www.facebook.com/RED21.com.au>






Confidentiality and Disclaimer:
The contents of this e-mail are confidential and are intended for the named recipient(s) only. If you are not the intended recipient you are hereby notified that any use, disclosure or distribution of the information contained in this e-mail is prohibited. If you have received this in error, please notify the sender immediately, and delete the e-mail and any attachments. RED21 has implemented anti-virus software and whilst all care is taken, it is the recipient’s responsibility to ensure that the message and any attachments are scanned for viruses prior to use.



________________________________

**Reply by Barbarur** [reply]
Hi @timcanning ,

Not sure if you added anything. Maybe a screenshot? Whatever you added, it's not visible on GitHub.

**Reply by timcanning** [reply]
![thumbnail_image](https://github.com/Barbarur/NovaPoint/assets/67358478/1e8af3a8-fe8d-4bf2-99ed-569443ace782)

Sorry, replied via email and image didn't come though.

FYI, when I check the error message. The file it mentions usually ( not 100% sure ) has no file version history, so it appears to process that file.
Also, just as I know this clients data is a bit of a mess, I have still been going to the folder in the error, copying and pasting the files to create a new copy, deleting the old and then restarting the NovaPoint process. This is how I have managed to get from 50k to 150k files, but it takes half a day to fail, so best case I get two runs a day and it is very slow progress.
If there was a way to start at file number 150K that would be the best way to progress this. But not sure whether that is possible.

Thanks and let me know if you need anything.

**Reply by Barbarur** [reply]
Hi @timcanning ,

That's interesting. Maybe that is a old migrated site with many forgotten files.

I see now where the issue is and I have some idea on how to avoid the solution to fully stop. Although I don't think I'll be able to get hands on it until next week.

Meanwhile, have you tried to add the server **Selative URL** for a specific folder in the Item Filter? You can go by batches adding the server relative URL i.e. *'/Sydney Share/TRADE, SERVICES & HOSTING/SERVICES/* to process only files on that folder, then move to the next main folder. Hopefully that save some time the next time you encounter this error.

**Reply by Barbarur** [reply]
Hi @timcanning ,

Can you test if the issue reproduces on the new version 0.7.0? I've implemented a step to avoid suddent finish of the solution when encountering corrupted file, but I don't have files with verison issue to test it.

**Comment by timcanning** [comment]
Thanks,

I am not sure if I am doing something wrong, but think I am picking all the same options.

Solution has started, please wait to the end
'DocumentLjbrary' 'Operational Team' is a large list with 289848 items. Expect the Solution to take
longer to run.
The 'query.FolderServerRelativeUrl' argument is invalid.

Thanks,



Tim Canning     Infrastructure Architect

[cid:Red21_logo_SIG2-Copy_7a1df1cb-0ccb-40d8-9f38-4933c30ab3bc.png]
RED21   Getting IT Right
WWW.RED21.COM.AU<http://www.RED21.com.au>

[cid:twitter60v4_738510bc-8a55-4c8e-8831-4c0f1d4c98e7.png]<http://twitter.com/red21au>  [cid:linkedin60v4_a7efd89b-d953-42de-b20f-1023b7178ee9.png] <http://www.linkedin.com/company/red21>   [cid:facebook60v4_76cebc4d-259e-4085-81ec-5281482d52ee.png] <http://www.facebook.com/RED21.com.au>






Confidentiality and Disclaimer:
The contents of this e-mail are confidential and are intended for the named recipient(s) only. If you are not the intended recipient you are hereby notified that any use, disclosure or distribution of the information contained in this e-mail is prohibited. If you have received this in error, please notify the sender immediately, and delete the e-mail and any attachments. RED21 has implemented anti-virus software and whilst all care is taken, it is the recipient’s responsibility to ensure that the message and any attachments are scanned for viruses prior to use.



________________________________

**Reply by Barbarur** [reply]
Can you check the logs file and share the parameters you are using?

You can check the wiki (https://github.com/Barbarur/NovaPoint/wiki/Definitions-files-and-items-filter) to see how the input should be for the server relative url.

**Comment by timcanning** [comment]
Delete all versions, single site, single library same as before. No filter.

Let me know if you need anything.


2024/06/02 04:24:50 - [NPLogger] - Logs: \RemoveFileVersionAuto_20240602042450\RemoveFileVersionAuto_20240602042450_Logs.txt
2024/06/02 04:24:50 - [NPLogger] - Report: \RemoveFileVersionAuto_20240602042450\RemoveFileVersionAuto_20240602042450_Report.csv
2024/06/02 04:24:50 - [NPLogger] - Solution properties
2024/06/02 04:24:50 - [NPLogger] - ========== ========== ==========
2024/06/02 04:24:50 - [NPLogger] - ReportMode: True
2024/06/02 04:24:50 - [NPLogger] - DeleteAll: True
2024/06/02 04:24:50 - [NPLogger] - VersionsToKeep: 500
2024/06/02 04:24:50 - [NPLogger] - Recycle: True
2024/06/02 04:24:50 - [NPLogger] - RemoveAdmin: True
2024/06/02 04:24:50 - [NPLogger] - AllSiteCollections: False
2024/06/02 04:24:50 - [NPLogger] - IncludePersonalSite: False
2024/06/02 04:24:50 - [NPLogger] - IncludeShareSite: False
2024/06/02 04:24:50 - [NPLogger] - OnlyGroupIdDefined: False
2024/06/02 04:24:50 - [NPLogger] - SiteUrl: https://XXXXXX.sharepoint.com
2024/06/02 04:24:50 - [NPLogger] - IncludeSubsites: False
2024/06/02 04:24:50 - [NPLogger] - ListOfSitesPath:
2024/06/02 04:24:50 - [NPLogger] - AllLists: False
2024/06/02 04:24:50 - [NPLogger] - IncludeLists: False
2024/06/02 04:24:50 - [NPLogger] - IncludeLibraries: True
2024/06/02 04:24:50 - [NPLogger] - IncludeHiddenLists: False
2024/06/02 04:24:50 - [NPLogger] - IncludeSystemLists: False
2024/06/02 04:24:50 - [NPLogger] - ListTitle: Operational Team
2024/06/02 04:24:50 - [NPLogger] - FolderRelativeUrl: /
2024/06/02 04:24:50 - [NPLogger] - ========== ========== ==========
2024/06/02 04:24:50 - [NPLogger] - Solution has started, please wait to the end
2024/06/02 04:24:50 - [AppInfo] - Adding cached token
2024/06/02 04:24:50 - [AppInfo] - Finish aquiring new Access Token with Refresh Token.
2024/06/02 04:24:50 - [AppInfo] - Access Token expiration time: 2/06/2024 5:42:33 AM +00:00
2024/06/02 04:24:50 - [GraphAPIHandler] - Request Message Uir https://graph.microsoft.com/v1.0/sites/root
2024/06/02 04:24:50 - [GraphAPIHandler] - Successful response
2024/06/02 04:24:50 - [Appinfo] - Hostname: XXXXXX.sharepoint.com
2024/06/02 04:24:50 - [Appinfo] - Domain: XXXXXX
2024/06/02 04:24:50 - [AppInfo] - SPO Admin URL 'https://XXXXXX-admin.sharepoint.com'
2024/06/02 04:24:50 - [AppInfo] - SPO -my URL 'https://XXXXXX-my.sharepoint.com'
2024/06/02 04:24:50 - [AppInfo] - SPO root URL 'https://XXXXXX.sharepoint.com'
2024/06/02 04:24:50 - [GetAADUser] - Getting Signed-in user
2024/06/02 04:24:50 - [AppInfo] - Got access token from memory
2024/06/02 04:24:50 - [AppInfo] - Access Token expiration time: 2/06/2024 5:42:33 AM +00:00
2024/06/02 04:24:50 - [GraphAPIHandler] - Request Message Uir https://graph.microsoft.com/v1.0/me
2024/06/02 04:24:51 - [GraphAPIHandler] - Successful response
2024/06/02 04:24:51 - [SPOTenantSiteUrlsCSOM] - Processing Site 'https://XXXXXX.sharepoint.com'
2024/06/02 04:24:51 - [SPOSiteCollectionAdminCSOM] - Processing 'https://XXXXXX.sharepoint.com' setting ***@***.***' IsSiteAdmin 'True'
2024/06/02 04:24:51 - [AppInfo] - Getting Access Token for root site https://XXXXXX-admin.sharepoint.com
2024/06/02 04:24:51 - [AppInfo] - Adding cached token
2024/06/02 04:24:51 - [AppInfo] - Finish aquiring new Access Token with Refresh Token.
2024/06/02 04:24:51 - [AppInfo] - Access Token expiration time: 2/06/2024 5:46:54 AM +00:00
2024/06/02 04:24:51 - [SPOListCSOM.Get] - Start getting Lists
2024/06/02 04:24:51 - [AppInfo] - Getting Access Token for root site https://XXXXXX.sharepoint.com
2024/06/02 04:24:51 - [AppInfo] - Adding cached token
2024/06/02 04:24:51 - [AppInfo] - Finish aquiring new Access Token with Refresh Token.
2024/06/02 04:24:51 - [AppInfo] - Access Token expiration time: 2/06/2024 4:33:04 AM +00:00
2024/06/02 04:24:51 - [SPOListCSOM] - Collected list 'Operational Team'
2024/06/02 04:24:51 - [SPOTenantListsCSOM] - Processing DocumentLibrary 'Operational Team'
2024/06/02 04:24:51 - [SPOTenantItemsCSOM] - Start Loop
2024/06/02 04:24:51 - [AppInfo] - Getting Access Token for root site https://XXXXXX.sharepoint.com
2024/06/02 04:24:51 - [AppInfo] - Adding cached token
2024/06/02 04:24:51 - [AppInfo] - Finish aquiring new Access Token with Refresh Token.
2024/06/02 04:24:51 - [AppInfo] - Access Token expiration time: 2/06/2024 4:33:04 AM +00:00
2024/06/02 04:25:03 - [SPOTenantItemsCSOM] - Collected from 'Operational Team' 5000 items...
2024/06/02 04:25:03 - [NPLogger] - Exception: The operation was canceled.
2024/06/02 04:25:03 - [NPLogger] - Trace: The operation was canceled.




Tim Canning     Infrastructure Architect

[cid:Red21_logo_SIG2-Copy_7a1df1cb-0ccb-40d8-9f38-4933c30ab3bc.png]
RED21   Getting IT Right
WWW.RED21.COM.AU<http://www.RED21.com.au>

[cid:twitter60v4_738510bc-8a55-4c8e-8831-4c0f1d4c98e7.png]<http://twitter.com/red21au>  [cid:linkedin60v4_a7efd89b-d953-42de-b20f-1023b7178ee9.png] <http://www.linkedin.com/company/red21>   [cid:facebook60v4_76cebc4d-259e-4085-81ec-5281482d52ee.png] <http://www.facebook.com/RED21.com.au>






Confidentiality and Disclaimer:
The contents of this e-mail are confidential and are intended for the named recipient(s) only. If you are not the intended recipient you are hereby notified that any use, disclosure or distribution of the information contained in this e-mail is prohibited. If you have received this in error, please notify the sender immediately, and delete the e-mail and any attachments. RED21 has implemented anti-virus software and whilst all care is taken, it is the recipient’s responsibility to ensure that the message and any attachments are scanned for viruses prior to use.



________________________________

**Reply by Barbarur** [reply]
Hi @timcanning ,

It seems the *FolderRelativeUrl* was forced  even when not used. It should be fixed now.
